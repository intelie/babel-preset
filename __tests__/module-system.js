import options from '../babel.config'
import { transformAsync } from '@babel/core'

describe('module system test suite', () => {
    test('commonjs', async () => {
        const input = `module.exports = 2`

        const { code: output } = await transformAsync(input, {
            ...options,
            filename: 'file.ts'
        })
        expect(output).toMatchSnapshot()

        const evaluated = eval(output)

        expect(evaluated).toEqual(2)
    })
})
