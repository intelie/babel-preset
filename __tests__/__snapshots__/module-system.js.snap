// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module system test suite commonjs 1`] = `
"\\"use strict\\";

module.exports = 2;"
`;
